Fixed chat service

HT-T opened the pull request:
Added chat service and chat function;
made sure that only 1 terminate collaboration button flags the collab session as "stop" instead of 2 separate buttons;
added interactive UI for better user experienceAdded chat service and chat function;
made sure that only 1 terminate collaboration button flags the collab session as "stop" instead of 2 separate buttons;
added interactive UI for better user experience

HT-T committed:
implemented chat and flag 'stop' for collab

chat function is implemented;
when a user stops collab, the session is flagged accordingly in firebase

Kidsnd274 committed:
Add Chat Service to docker-compose and fix Dockfile issues

Kidsnd274 committed:
Update Dockerfile and run.py

Dockerfile now copies firebase creds and .env
run.py now checks for env variable to decide whether to use waitress or run normally
Use waitress for production environment

HT-T committed:
change /collab to /collaboration

Kidsnd274 committed:
Make chat into component

Kidsnd274 committed:
Merge branch 'main' into chat

# Conflicts:
#	frontend/Peerprep/package-lock.json
#	frontend/Peerprep/pages/collaboration/index.vue

Kidsnd274 committed:
Fix package-lock

Kidsnd274 committed:
Add Chat component

HT-T committed:
fix index.vue

HT-T committed:
fixed chat function, used collab ID

chat function fixed after rebasing code; also used the original collaboration ID to distinguish different chats for users

HT-T committed:
Unify "stop collab" buttons

HT-T committed:
Add CSS style to convo selection

Added interactive UI when users selecting a conversation

HT-T committed:
Delete index.vue

Delete an extra unused index.vue

HT-T committed:
Merge branch 'main' into chat

Huggenguggen committed:
change docker compose workflow

Add chat-service build env

HT-T committed:
Minor fixes

Removed collab from navbar;
user now jumps to collab page automatically;
removed duplicated index.vue

HT-T committed:
Merge branch 'chat' of https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g29 into chat

HT-T committed:
Optimized chat

User now auto jumped to newest active chat when entering a collab session

HT-T committed:
Optimized chat

introduced delay to make sure new convo is loaded;
delete redundant frontend/collab file

Huggenguggen committed:
clean up Navbar.vue

Remove unnecessary spacing

yfg-dev reviewed (commented):
![image](https://github.com/user-attachments/assets/1bfaa6df-80a1-4a56-9ce9-743e2bdebd5d)
with the latest code, not sure why this error occurs for one of the account. unable to press submit and the id not displayed

the other user seems to be working fine.
![image](https://github.com/user-attachments/assets/7e608763-b1b9-4f0a-97e4-8486b21e3024)

after refrshing the page for the user with error, seems to be working but there is still error message displayed in the console
![image](https://github.com/user-attachments/assets/eda07622-72e5-4aa9-b440-9260f03d3d24)


the first error does not exist after logout and relogin. hope that's one off incident

yfg-dev committed:
Clean up collaborate and chat components for deployment

HT-T committed:
timezone issues fixed

Time stamps now display in local time

HT-T committed:
Merge branch 'chat' of https://github.com/CS3219-AY2425S1/cs3219-ay2425s1-project-g29 into chat

HT-T committed:
optimize chat list loading

Introduced waiting to absolutely make sure the first user does not jump to the collab page too fast and fail to see the CONVERSATION

Huggenguggen reviewed (approved):
LGTM
